### PR TITLE
Compatibility with the Changes in Cblox

### PR DESCRIPTION
**Pull request by gasserl**
C-blox recently got a significant update, this PR aims to minimize redundancy between the two repos.

**Main Changes**
- Added dependency on cblox_msgs and cblox_ros
- Using messages and serialization functions from cblox
- Using mission frame name from ros parameters
- Changed function arguments from const ref submap to const ref submap shared ptr for consistency

### CODE REVIEW

File: voxgraph/include/voxgraph/tools/data_servers/submap_server.h [hunk 1]
```diff
@@ -2,24 +2,25 @@
...
-  void publishSubmapTsdf(const VoxgraphSubmap& submap,
+  void publishSubmap(const VoxgraphSubmap::Ptr& submap_ptr, const ros::Time& timestamp);
+  void publishSubmapTsdf(const VoxgraphSubmap::Ptr& submap_ptr,
                          const ros::Time& timestamp);
-  void publishSubmapTsdfAndEsdf(const VoxgraphSubmap& submap,
+  void publishSubmapTsdfAndEsdf(const VoxgraphSubmap::Ptr& submap_ptr,
                                 const ros::Time& timestamp);
-  void publishSubmapSurfacePointcloud(const VoxgraphSubmap& submap,
+  void publishSubmapSurfacePointcloud(const VoxgraphSubmap::Ptr& submap_ptr,
                                       const ros::Time& timestamp);
```
Comment (alexmillane) [inline voxgraph/include/voxgraph/tools/data_servers/submap_server.h:24]: Why change to the pointers here?

File: voxgraph/include/voxgraph/tools/data_servers/submap_server.h [hunk 2]
```diff
@@ -54,11 +58,7 @@ class SubmapServer {
 
   static constexpr bool fake_6dof_transforms_ = true;
 
-  // Convenience methods to generate the message and submap headers
-  static std_msgs::Header generateHeaderMsg(const VoxgraphSubmap& submap,
-                                            const ros::Time& timestamp);
-  static voxgraph_msgs::MapHeader generateSubmapHeaderMsg(
-      const VoxgraphSubmap& submap);
+  FrameNames frame_names_;
```
Comment (alexmillane) [inline voxgraph/include/voxgraph/tools/data_servers/submap_server.h:61]: Great, thanks for this.

File: voxgraph/src/frontend/submap_collection/voxgraph_submap.cpp [hunk 3]
```diff
@@ -93,6 +93,10 @@ void VoxgraphSubmap::finishSubmap() {
   std::cout << "\n# isosurface vertices: " << isosurface_vertices_.size()
             << std::endl;
 
+  // Set mapping times
+  mapping_interval_.first = getStartTime().toSec();
+  mapping_interval_.second = getEndTime().toSec();
```
Comment (alexmillane) [inline voxgraph/src/frontend/submap_collection/voxgraph_submap.cpp:98]: Is there a reason to use seconds here? I think we should change cblox to store `uint64_t` and use nano seconds (see [here](http://docs.ros.org/latest/api/rostime/html/classros_1_1TimeBase.html#ac6c1cb1ef005601de453a14cace6be26)).

Comment (victorreijgwart) [reply voxgraph/src/frontend/submap_collection/voxgraph_submap.cpp:98]: +1
nanoseconds in `uint64_t` would be more general and cover all possible use cases :)

Comment (gasserl) [reply voxgraph/src/frontend/submap_collection/voxgraph_submap.cpp]: Agreed, I've changed it here and in cblox in this [PR](https://github.com/ethz-asl/cblox/pull/34)

File: voxgraph/include/voxgraph/tools/data_servers/submap_server.h (hunk 1 above)
Comment (gasserl) [reply voxgraph/include/voxgraph/tools/data_servers/submap_server.h:24]: The underlying conversion functions from cblox take the pointer as an argument (see [here](https://github.com/ethz-asl/cblox/blob/master/cblox_ros/include/cblox_ros/submap_conversions.h#L18)), so it seemed like the easier option to me. I could also generate a pointer in the function or change the cblox functions to take a const ref, as you prefer.

Comment (alexmillane) [reply voxgraph/include/voxgraph/tools/data_servers/submap_server.h:24]: If possible could we change the underlying functions in cblox to take const refs? The problem, in my view, with taking shared_ptrs in contexts like this is that it forces the caller to have a shared pointer as opposed to a reference, and this requirement propagates all the way up the call tree. 

Just to make sure we're not totally off here, I checked voxblox, and [they also take const refs](https://github.com/ethz-asl/voxblox/blob/master/voxblox_ros/include/voxblox_ros/conversions.h#L148).

Comment (gasserl) [reply voxgraph/include/voxgraph/tools/data_servers/submap_server.h]: Everything's back to const refs and the cblox update is in this [PR](https://github.com/ethz-asl/cblox/pull/35).

**Pull request merged**